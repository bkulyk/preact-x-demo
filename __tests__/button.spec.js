import React from 'react';
import { shallow } from './enzyme';
import Button from '@bitchin/button';
import sinon from 'sinon';

it('renders correctly', () => {
  expect(shallow(<Button>Facebook</Button>))
    .toMatchSnapshot();
});

it('renders a secondary button', () => {
  expect(shallow(<Button secondary>Facebook</Button>))
    .toMatchSnapshot();
});

it('disabled button should render correctly', () => {
  const handleClick = sinon.spy();
  const rendered = shallow((
    <Button disabled onClick={handleClick}>Facebook</Button>
  ));

  expect(rendered).toMatchSnapshot();
});

it('outlined button should render correctly', () => {
  const rendered = shallow((
    <Button outlined>Facebook</Button>
  ));
  expect(rendered).toMatchSnapshot();
});

it('link button should render correctly', () => {
  const rendered = shallow((
    <Button href="#facebook">Facebook</Button>
  ));
  expect(rendered).toMatchSnapshot();
});

it('should allow click events', () => {
  const handleClick = sinon.spy();
  const rendered = shallow(<Button onClick={handleClick}>Facebook</Button>);

  expect(rendered).toMatchSnapshot();

  rendered.find('button').simulate('click');

  expect(handleClick.called).toEqual(true);
});