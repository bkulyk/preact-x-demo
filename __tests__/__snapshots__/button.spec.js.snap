// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`disabled button should render correctly 1`] = `
<Ripple>
  <button
    className="mdc-button mdc-button--raised"
    disabled={true}
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    <span
      className="mdc-button__label"
    >
      Facebook
    </span>
  </button>
</Ripple>
`;

exports[`link button should render correctly 1`] = `
<Ripple>
  <button
    className="mdc-button"
    disabled={false}
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    <span
      className="mdc-button__label"
    >
      Facebook
    </span>
  </button>
</Ripple>
`;

exports[`outlined button should render correctly 1`] = `
<Ripple>
  <button
    className="mdc-button mdc-button--outlined"
    disabled={false}
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    <span
      className="mdc-button__label"
    >
      Facebook
    </span>
  </button>
</Ripple>
`;

exports[`renders a secondary button 1`] = `
<Ripple>
  <button
    className="mdc-button mdc-button--raised mdc-theme--secondary-bg mdc-theme--on-secondary mdc-theme--secondary"
    disabled={false}
    onClick={[Function]}
    style={
      Object {
        "borderColor": "var(--mdc-theme-secondary, #018786)",
      }
    }
    type="button"
  >
    <span
      className="mdc-button__label"
    >
      Facebook
    </span>
  </button>
</Ripple>
`;

exports[`renders correctly 1`] = `
<Ripple>
  <button
    className="mdc-button mdc-button--raised"
    disabled={false}
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    <span
      className="mdc-button__label"
    >
      Facebook
    </span>
  </button>
</Ripple>
`;

exports[`should allow click events 1`] = `
<Ripple>
  <button
    className="mdc-button mdc-button--raised"
    disabled={false}
    onClick={[Function]}
    style={Object {}}
    type="button"
  >
    <span
      className="mdc-button__label"
    >
      Facebook
    </span>
  </button>
</Ripple>
`;
